rfcs/0000-unified-configuration.ftd

```diff
@@ -0,0 +1,312 @@
+-- import: fastn.com/rfcs/lib
+-- import: fastn-community.github.io/bling/note
+
+-- lib.rfc: RFC: Unified Configuration
+id: unified-config
+status: proposal
+
+A configuration system streamlining complex settings for processors, fastn
+packages. It introduces:
+
+- Special Configuration types in the `fastn` namespace.
+- A virtual module for configuration access in userland.
+- Package level configuration.
+
+-- lib.motivation:
+
+Currently, `fastn` supports configuration through [environment
+variables](/env/), cli arguments, and in some way, through the `FASTN.ftd`
+file. Environment variables are good to change the behaviour of the cli or to
+pass secret information but it's not ideal for defining complex configuration
+for a framework like `fastn` which provides various configurable features like
+[processors](/processor/-/backend/), oauth authentication and, apps.
+
+This RFC proposes a way of dealing with configuration that is centralized and
+leverages the existing ability of the `fastn` cli to read environment
+variables.
+
+-- end: lib.motivation
+
+
+-- lib.detailed-design:
+
+-- ds.h2: Configuration types in `fastn.*`
```

Comment:
We can omit this part of the proposal and focus on how to define package configuration variable and how to access them, and the special module only. This is to reduce the scope of this RFC and we can create more RFC for the connection types. For now we only need DATABASE_URL and other such things, which are just strings.